rest_api/sawtooth_rest_api/route_handlers.py

```diff
@@ -262,7 +262,7 @@ def __init__(
             paging: Paging info and nav, like total resources and a next link
         """
         paging_controls = self._get_paging_controls(request)
-
+        head_id=self._get_head_id(request)
```

Comment:
This should be pushed down into the `_head_to_root` function - this particular error can occur in the `fetch_state` function as well.

Likewise, as that head_id is unused, I suspect the build is failing due to lint

Reply:
Thanks for suggesting _head_to_root function.

PR specific to list_state function (for endpoint '/state'). Please refer "start_rest_api" in rest_api.py, fetch_state for '/state/{address}' endpoint.

As per my understanding, ClientStateListRequest is only function where head_id not pass as parameter in comparison to other functions (ClientTransactionListRequest, ClientBatchListRequest and ClientBlockListRequest).

message ClientStateListRequest {
	string state_root = 1;
	string address = 3;
	ClientPagingControls paging = 4;
	repeated ClientSortControls sorting = 5;
}

message ClientBlockListRequest {
	string head_id = 1;
	repeated string block_ids = 2;
	ClientPagingControls paging = 3;
	repeated ClientSortControls sorting = 4;
}

Adding head_id in ClientStateListRequest, impact many other functionality. Please suggest.

![clientblocklistrequest](https://user-images.githubusercontent.com/35533098/45486092-7f53f500-b777-11e8-9be1-729ebf5a62c3.PNG)